### ISSUE TITLE
confused about documentation's data system examples

### ISSUE THREAD

**Issue by lingtalfi** [Polymer/docs#1730]
tl;dr
Could you please make examples that assume that the reader knows nothing about Polymer, except how to include it in her project?

Trying to read this section:
https://www.polymer-project.org/1.0/docs/devguide/data-system#data-flow-examples

and specifically to understand the comments in the code of the first example (Two-way binding):

``` html
<script>
  Polymer({
    is: 'custom-element',
    properties: {
      someProp: {
        type: String,
        notify: true
      }
    }
  });
</script>
...

<!-- changes to "value" propagate downward to "someProp" on child -->
<!-- changes to "someProp" propagate upward to "value" on host  -->
<custom-element some-prop="{{value}}"></custom-element>
```

Here is my test rig, but apparently I totally don't understand what's supposed to happen:

``` html
<dom-module id="custom-element">
    <template>
        <div>Hello {{someProp}}</div>
    </template>
</dom-module>
<script>


    Polymer({
        is: 'custom-element',
        properties: {
            someProp: {
                type: String,
                notify: true
            }
        }
    });
</script>


<!-- changes to "value" propagate downward to "someProp" on child -->
<!-- changes to "someProp" propagate upward to "value" on host  -->
<custom-element some-prop="{{value}}"></custom-element>


<script>
    // trying to change value and see if it propagates downward to "someProp" on child: nope
    // expected: the text content of the div should be 'Hello ho', but it's 'Hello {{value}}'
    el.value = "ho";


    // trying to change someProp and see if it propagates upward to "value" on host: nope either
    // expected: console.log(el) should give: <custom-element some-prop="jj"></custom-element>
    // but it actually gives: <custom-element some-prop="{{value}}"></custom-element>
    el.someProp = "jj";

</script>
```

Could you provide full working examples, as foreign users like me may want to understand a specific section of the documentation without reading through the whole docs first, and having a working sandbox code is sometimes/often more explicit than the documentation itself.

**Comment by arthurevans** [comment]
Agreed, these samples are confusing because they are missing the context of the host element.

**Comment by arthurevans** [comment]
For 2.0, we revised all of these examples and included the full context. 

https://www.polymer-project.org/2.0/docs/devguide/data-system#data-flow-examples

Hope these help!
